### ISSUE TITLE
Incorrect extension block params nullability warning in .NET 10

### ISSUE THREAD

**Issue by jeremia** [dotnet/roslyn#81394]
**Version Used**: 
dotnet 10.0.100

**Steps to Reproduce**:

```csharp
public class TestClass
{
    public void DoSomething(string arg)
    {
        this.DoExtension(arg);
    }
}

public static class TestExtensions
{
    extension(TestClass testClass){
        public void DoExtension(params string[] args)
        {
        }
    }
}
```

The following code does not result in a warning:
```csharp
public class TestClass
{
    public void DoSomething(string arg)
    {
        this.DoExtension(arg);
    }
}

public static class TestExtensions
{
    public static void DoExtension(this TestClass testClass, params string[] args)
    {
    }
}
```

As far as I can see .NET Lab doesn't show compiler warnings, so I didn't provide a repo link there.

**Diagnostic Id**:
 warning CS8620: Argument of type 'string' cannot be used for parameter 'args' of type 'string[]' in 'void extension(TestClass).DoExtension(params string[] args)' due to differences in the nullability of reference types.

**Expected Behavior**:
No warning

**Actual Behavior**:
A warning is given

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/81205
- https://github.com/dotnet/roslyn/issues/80320
- https://github.com/dotnet/roslyn/issues/80024
